drizzle-orm/src/aws-data-api/pg/session.ts

```diff
@@ -89,6 +93,18 @@ export class AwsDataApiPreparedQuery<T extends PreparedQueryConfig> extends Prep
 			return row.map((field: Field) => getValueFromDataApi(field));
 		});
 	}
+
+	/** @internal */
+	mapResultRows(records: Field[][], columnMetadata: ColumnMetadata[]) {
+		return records.map((record) => {
+			const row: Record<string, unknown> = {};
+			for (const [index, field] of record.entries()) {
+				const { name } = columnMetadata[index]!;
+				row[name || index] = getValueFromDataApi(field); // not what to default if name is undefined
```

Comment:
Any reason you've used `||` here instead of `??` ?

Reply:
As I've added as a comment, I'm not entirely sure what the default behaviour should be if the column key is not found. But it made sense to default it to the index if/when it's undefined. 
I've used two pipes for the general reason that it would default to the index, if the value on the left wouldn't be a valid array index, but as the values we're expecting are either string or undefined, I'm happy to do a nullish coalescing operation as well